modules/tinymce/src/core/test/ts/browser/options/SidebarShowOptionTest.ts

```diff
@@ -0,0 +1,87 @@
+
+import { Waiter } from '@ephox/agar';
+import { TestHelpers } from '@ephox/alloy';
+import {describe, it } from '@ephox/bedrock-client';
+import { Sidebar } from '@ephox/bridge';
+import { SugarElement, Traverse } from '@ephox/sugar';
+import { McEditor } from '@ephox/wrap-mcagar';
+
+interface EventLog {
+  readonly name: string;
+  readonly index: number;
+}
+
+describe('browser.tinymce.core.options.sidebar_show', () => {
+  const settingsFactory = (store: TestHelpers.TestStore) => {
+    const logEvent = (name: string) => (api: Sidebar.SidebarInstanceApi) => {
+      const index = Traverse.findIndex(SugarElement.fromDom(api.element())).getOr(-1);
+      const entry: EventLog = { name, index };
+      store.adder(entry)();
+    };
+
+    return {
+      base_url: '/project/tinymce/js/tinymce',
+      setup: (ed) => {
+        ed.ui.registry.addSidebar('sidebarone', {
+          tooltip: 'side bar one',
+          icon: 'comment',
+          onShow: logEvent('sidebarone:show'),
+        });
+        ed.ui.registry.addSidebar('sidebartwo', {
+          tooltip: 'side bar one',
+          icon: 'comment',
+          onShow: logEvent('sidebartwo:show'),
+        });
+      }
+    };
+  };
+
+  it('TINY-8710: Show no sidebar on init if not set', async () => {
+    const store = TestHelpers.TestStore();
```

Comment:
This probably doesn't need to be initialised for each test. We could init this outside and then just do a `beforeEach` that calls `clear()` to reset the store. That'd just remove some duplication here is all.